File: app/src/androidTest/java/org/schabi/newpipe/report/ErrorInfoTest.java
package org.schabi.newpipe.report;

import android.os.Parcel;
import android.support.test.filters.LargeTest;
import android.support.test.runner.AndroidJUnit4;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.schabi.newpipe.R;
import org.schabi.newpipe.report.ErrorActivity.ErrorInfo;

import static org.junit.Assert.assertEquals;

/**
 * Instrumented tests for {@link ErrorInfo}
 */
@RunWith(AndroidJUnit4.class)
@LargeTest
public class ErrorInfoTest {

    @Test
    public void errorInfo_testParcelable() {
        ErrorInfo info = ErrorInfo.Companion.make(UserAction.USER_REPORT, "youtube", "request", R.string.general_error);
        // Obtain a Parcel object and write the parcelable object to it:
        Parcel parcel = Parcel.obtain();
        info.writeToParcel(parcel, 0);
        parcel.setDataPosition(0);
        ErrorInfo infoFromParcel = ErrorInfo.Companion.getCREATOR().createFromParcel(parcel);

        assertEquals(UserAction.USER_REPORT, infoFromParcel.getUserAction());
        assertEquals("youtube", infoFromParcel.getServiceName());
        assertEquals("request", infoFromParcel.getRequest());
        assertEquals(R.string.general_error, infoFromParcel.getMessage());

        parcel.recycle();
    }
}

File: app/src/androidTest/java/org/schabi/newpipe/report/ErrorInfoTest.kt
package org.schabi.newpipe.report

import android.os.Parcel
import androidx.test.InstrumentationRegistry
import androidx.test.filters.LargeTest
import androidx.test.runner.AndroidJUnit4
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.schabi.newpipe.R

/**
 * Instrumented tests for [ErrorInfo]
 */
@RunWith(AndroidJUnit4::class)
@LargeTest
class ErrorInfoTest {

    private val context = InstrumentationRegistry.getContext()

    @Test
    fun errorInfo_testParcelable() {
        val info = ErrorInfo.make(UserAction.USER_REPORT, "youtube", "request", R.string.general_error)
        // Obtain a Parcel object and write the parcelable object to it:
        val parcel = Parcel.obtain()
        info.writeToParcel(parcel, 0)
        parcel.setDataPosition(0)

        val infoFromParcel = ErrorInfo.CREATOR.createFromParcel(parcel)

        assertEquals(UserAction.USER_REPORT, infoFromParcel.userAction)
        assertEquals("youtube", infoFromParcel.serviceName)
        assertEquals("request", infoFromParcel.request)
        assertEquals(R.string.general_error.toLong(), infoFromParcel.message.toLong())

        parcel.recycle()
    }
}